### PR TITLE
tweaks to #1027 - single more general protocol

### PR DESCRIPTION
**Pull request by JoranAngevaare**
This PR just tweaks the protocols in #1027 to make it a bit more general. Core idea is still @ershockley 's 

## Differences
 - Only difference is that now the protocol would return the keys as they are requested from the json`'gain_ab'` instead of `'ab'` - depending on what the columns are in the file.
 - Call the time column `'time'` instead of `'times'` or `'timestamps'`

## Updates
- I've also added a few words of documentation and also expanded the test to work on CSV files 
- testing code on the CorrectedAreas-plugin

## Further generalization options
 - Not sure if the comma separation `keys = strax.to_str_tuple(itp_dict_keys.split(','))` could be handled by the URLConfig

### CODE REVIEW

File: straxen/plugins/event_processing.py [hunk 1]
```diff
@@ -785,10 +785,12 @@ def compute(self, events):
...
-            cs2_top_xycorr = (events[f'{peak_type}s2_area'] * events[f'{peak_type}s2_area_fraction_top'] / self.s2_xy_map(s2_positions, map_name=s2_top_map_name))
-            cs2_bottom_xycorr = (events[f'{peak_type}s2_area'] *
-                                       (1 - events[f'{peak_type}s2_area_fraction_top']) /
-                                       self.s2_xy_map(s2_positions, map_name=s2_bottom_map_name))
+            cs2_top_xycorr = (events[f'{peak_type}s2_area']
+                              * events[f'{peak_type}s2_area_fraction_top']
+                              / self.s2_xy_map(s2_positions, map_name=s2_top_map_name))
+            cs2_bottom_xycorr = (events[f'{peak_type}s2_area']
+                                 * (1 - events[f'{peak_type}s2_area_fraction_top'])
+                                 / self.s2_xy_map(s2_positions, map_name=s2_bottom_map_name))
```
Comment (JoranAngevaare) [inline straxen/plugins/event_processing.py:793]: Hmm, this wasn't intended to go into this PR, nevertheless, we try not to make lines as long as this one so probably doesn't hurt adding this change. The other tweak is fixing https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

> `cs2_top_xycorr = (events[f'{peak_type}s2_area'] * events[f'{peak_type}s2_area_fraction_top'] / self.s2_xy_map(s2_positions, map_name=s2_top_map_name))`

File: straxen/url_config.py [hunk 2]
```diff
@@ -335,57 +336,45 @@ def open_neural_net(model_path: str, **kwargs):
...
-        if is_partition:
-            interp_ab = interp1d(times, loaded_json['ee_ab'], bounds_error=True)
-            interp_cd = interp1d(times, loaded_json['ee_cd'], bounds_error=True)
-            gain_ab = interp_ab(start)
-            gain_cd = interp_cd(start)
-            return {'ab': gain_ab, 'cd': gain_cd}
+        if len(strax.to_str_tuple(keys))>1:
+            return {key:
+                    interp1d(times, loaded_json[key], bounds_error=True)(start)
+                    for key in keys}
```
Comment (JoranAngevaare) [inline straxen/url_config.py:373]: The difference: used to return `{'ab': gain_ab, 'cd': gain_cd}`, now returns `{'gain_ab': gain_ab, 'gain_cd': gain_cd}` (depending on how the json looks)

**Comment by ershockley** [comment]
@JoranAngevaare I made a few changes to make the protocol more generalizable (allows for passing time key as a parameter) in order for it to be backwards compatible. I also had to add an additional protocol that reformats the dict to be what the plugin expects, again for backwards compatibility. 

Personally I still feel like the other protocols are a bit more transparent, even if they aren't as general as these ones, as now the urls get crazy long. But if you prefer this way we can go with this. Let me know what you think.

**Comment by ershockley** [comment]
I've changed my mind, let's go with the general protocols but with longer urls. Probably better for longterm.

**Review by ershockley — APPROVED** [review]

**Pull request merged**